### PR TITLE
Add introductory information to debugging index

### PR DESCRIPTION
**Pull request by GawainLynch**
After this is merged and cascade, I'll PR 3.3 with the recent changes to core by @CarsonF

### CODE REVIEW

**Comment by bobdenotter** [comment]
Looks good. Perhaps change this: 

```
It is also advised to set the debug error level to show all errors:

    debug_error_level: -1

``` 

to: 

```
It is also advised to set the debug error level to show all errors, as well as set Twig to be strict on the use of undeclared variables:

    debug_error_level: -1
    strict_variables: true
```

?

File: docs/debugging/index.md [hunk 1]
```diff
@@ -9,4 +9,106 @@ pages:
...
+
+#### Basic
+
+For very basic debugging on PHP problems, Bolt has its own [`dump`][dump-php]
+and [`backtrace`][backtrace-php] functions available.
+
+#### Advanced
+
+However, while more complex to set up, using `xdebug` in an IDEs such as
+[PhpStorm][xdebug-phpstorm], [Atom][xdebug-atom] or [Eclipse][xdebug-eclipse] will
```
Comment (bobdenotter) [inline docs/debugging/index.md]: , [VSCode][xdebug-vscode]

[xdebug-vscode]: https://github.com/felixfbecker/vscode-php-debug

File: docs/debugging/index.md [hunk 2]
```diff
@@ -9,4 +9,106 @@ pages:
...
+  * Database queries that were performed for the page load
+
+#### Basic
+
+For very basic debugging on PHP problems, Bolt has its own [`dump`][dump-php]
+and [`backtrace`][backtrace-php] functions available.
+
+#### Advanced
+
+However, while more complex to set up, using `xdebug` in an IDEs such as
```
Comment (bobdenotter) [inline docs/debugging/index.md]: IDE, singular.

**Comment by GawainLynch** [comment]
How about now. I put the `strict_variables` in the Twig section, as to be fair, that's topically where is makes the most sense.

**Pull request merged**